src/GraphQL/Types/QueryArguments.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace GraphQL.Types
 {
-    public class QueryArguments : IEnumerable<QueryArgument>
+    public class QueryArguments : IReadOnlyCollection<QueryArgument>, ICollection
```

Comment:
with this change doe we still need `QueryArguments` ?

Reply:
I asked this question from time to time even before these changes.

Reply:
It's a read-only collection and supports backwards compatibility.  If we are not going to pursue changing the design to utilize read-only interfaces, I'd rather use the `new[] { ... }` syntax (currently not supported due to field builders requiring a `QueryArguments` class).  An array isn't read-only, however.